hive-hooks/path-conversion-hook/src/main/java/com/expediagroup/apiary/extensions/hooks/pathconversion/config/Configuration.java

```diff
@@ -0,0 +1,93 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.apiary.extensions.hooks.pathconversion.config;
+
+import static java.lang.String.format;
+
+import java.util.*;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang.BooleanUtils;
+import org.apache.hadoop.hive.conf.HiveConf;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import com.google.common.collect.ImmutableList;
+
+import com.expediagroup.apiary.extensions.hooks.pathconversion.models.PathConversion;
+
+@Slf4j
+@Getter
+public class Configuration {
+    public static final String PATH_REPLACEMENT_PREFIX = "apiary.path.replacement";
+    public static final String PATH_REPLACEMENT_ENABLED = format("%s.enabled", PATH_REPLACEMENT_PREFIX);
+    public static final String PATH_REPLACEMENT_REGEX = format("%s.regex", PATH_REPLACEMENT_PREFIX);
+    public static final String PATH_REPLACEMENT_VALUES = format("%s.value", PATH_REPLACEMENT_PREFIX);
+    public static final String PATH_REPLACEMENT_GROUPS = format("%s.capturegroups", PATH_REPLACEMENT_PREFIX);
+
+    private final Properties properties;
+
+    private final boolean pathConversionEnabled;
+    private final List<PathConversion> pathConversions;
+
+    public Configuration(HiveConf conf) {
+        Properties props = conf.getAllProperties();
+
+        if (Objects.isNull(props)) {
+            log.warn("Could not load properties from HiveConf! Hive may be mis-configured.");
+            props = new Properties();
+        }
+
+        properties = props;
+        pathConversionEnabled = BooleanUtils.toBoolean(props.getProperty(PATH_REPLACEMENT_ENABLED, "false"));
+        pathConversions = initializePathReplacements();
+    }
+
+    private List<PathConversion> initializePathReplacements() {
+        List<PathConversion> pathConversions = new ArrayList<>();
+        for (Object keyObj : properties.keySet()) {
+            String curPropName = (String) keyObj;
+
+            if (curPropName.startsWith(PATH_REPLACEMENT_REGEX)) {
+                String valuePropName = curPropName.replace(PATH_REPLACEMENT_REGEX, PATH_REPLACEMENT_VALUES);
+                String value =  properties.getProperty(valuePropName);
+
+                String captureGroupPropName = curPropName.replace(PATH_REPLACEMENT_REGEX, PATH_REPLACEMENT_GROUPS);
+                List<Integer> captureGroups = getCaptureGroups(captureGroupPropName);
+
+                if (Objects.isNull(value)) {
```

Comment:
You can move this to line 69 can't you? There's no point figuring out the capture groups if it's null right?